### DISCUSSION TITLE
Use hex color for badges

### DISCUSSION THREAD

**Discussion by lotarbo** [filamentphp/filament#12684] [Help]
### Package

Table builder

### Package Version

v.3.2.73

### How can we help you?

Hi. How can I use hex for badge?
Ex: 
```php
TextColumn::make('status.name')->badge()->color(fn(Contact $record): ?string => $record->status->color),
``` 
Where color contains hex value, `#EDFF21`

**Comment by lotarbo** [comment]
Right now I can only see expanding the colors in this way.
if AdminPanelProvider.php
```php
$colors = Status::whereNotNull('color')->pluck('color', 'color')->mapWithKeys(fn($color, $key) => [Str::remove('#',  $color) => Color::hex($color)])->toArray();
``` 
and
```php
$panel->colors($colors);
``` 

And it's working.  But it seems kind of complicated.

**Comment by dan-pascu** [comment]
Hi,

This worked for me:

Tables\Columns\TextColumn::make('status.value')
                        ->color(function ($record) { return Color::hex($record->status->color); })
                        ->badge(),
